src/commands/ext-dev-publish.ts

```diff
@@ -45,15 +58,33 @@ export const command = new Command("ext:dev:publish <extensionRef>")
         )}'. Please use the format '${clc.bold("<publisherId>/<extensionId>")}'.`
       );
     }
-    const extensionYamlDirectory = findExtensionYaml(process.cwd());
-    const res = await publishExtensionVersionFromLocalSource({
-      publisherId,
-      extensionId,
-      rootDirectory: extensionYamlDirectory,
-      nonInteractive: options.nonInteractive,
-      force: options.force,
-      stage: options.stage ?? "stable",
-    });
+    if ((options.repo || options.root) && !options.ref) {
+      throw new FirebaseError("`ref` must be set if `repo` or `root` are set.");
```

Comment:
Should this error or just set ref to HEAD?

Reply:
Good point, done. Changed this condition to default to new path if any of the arguments are set, and it'll default to HEAD later when it prompts.